crates/sources/src/loader.rs

```diff
@@ -111,11 +118,13 @@ impl<F: Fetcher> Loader<F> {
         // At this point we know that no more fetches will complete.
         // Re-write capture and materialization configurations to their inline form.
 
-        let to_inline = |endpoint_spec: &mut serde_json::Value| {
-            let taken = serde_json::from_value(std::mem::take(endpoint_spec))
+        let to_inline = |endpoint_spec: &mut Box<RawValue>| {
+            let taken = serde_json::from_str(endpoint_spec.get())
                 .expect("endpoint spec must be a ConnectorConfig");
 
-            *endpoint_spec = match taken {
+            *endpoint_spec = RawValue::from_string("null".to_owned()).unwrap();
```

Comment:
This replaces the previous `std::mem::take` which would set a default value. There is no default value for `RawValue` AFAIK.